api/webhook/affected_namespaces.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1/genericwebhook"
+)
+
+func affectedNamespaces(obj interface{}) (bool, map[string]struct{}) {
+	clusterScoped := false
+	namespaces := make(map[string]struct{})
+
+	walker := genericwebhook.NewFieldWalker(obj, func(path *field.Path, obj interface{}, field *reflect.StructField) bool {
+
+		// These are some trivial rules to cut a lot of edges.
+		if field != nil && (field.Name == "Status" || field.Name == "TypeMeta" || field.Name == "ObjectMeta") {
+			return false
+		}
+
+		if selector, ok := obj.(*v1alpha1.PodSelector); ok {
+			if selector == nil {
+				return false
+			}
+
+			for _, ns := range selector.Selector.Namespaces {
```

Comment:
how about reusing the function  `AffectedNamespaces` https://github.com/chaos-mesh/chaos-mesh/blob/1ca8c59d84d6152838b98d08553fe66a85d6d728/api/v1alpha1/selector.go#L133

Reply:
Oops, I have ignored the `Pods`. As this function is not used in any other places, I moved it here directly (inline).